R/locate_data.R

```diff
@@ -0,0 +1,48 @@
+#' Locate local data files
+#'
+#' @inheritParams load_local_vimc_climate
+#'
+#' @param country_iso3c Country ISO 3 character code. Passed from top-level
+#' function [load_local_vimc_climate()] after a call to [validate_country()].
+#'
+#' @keywords internal data_locator
+#'
+#' @return The location of a local data Parquet file. Has a number of
+#' side-effects to inform users when `data_location`, the county data directory
+#' within `data_location`, or the file corresponding to the data requested are
+#' not found.
+locate_parquet_file <- function(
+  data_location,
+  country_iso3c,
+  admin_level,
+  data_source
+) {
+  checkmate::assert_directory_exists(
+    data_location
+  )
+  data_location <- file.path(data_location, country_iso3c)
+  checkmate::assert_directory_exists(
+    data_location
+  )
+
```

Comment:
You can use `assert_file_exists` and `assert_is_directory` here, or update reside.utils to offer `assert_directory_exists` by wrapping these together if you want

Reply:
I'll write package specific validators for the data location and data file. I think `assert_is_directory` satisfies the use case here so don't think it's worth adding a new assertion function.